File: vipclaw-admin/src/main/java/com/vipclaw/admin/dto/LoginRequest.java
package com.vipclaw.admin.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

/**
 * 登录请求 DTO
 */
@Data
@Schema(description = "登录请求对象")
public class LoginRequest {

    @Schema(description = "用户名", example = "admin", requiredMode = Schema.RequiredMode.REQUIRED)
    private String username;

    @Schema(description = "密码", example = "admin123", requiredMode = Schema.RequiredMode.REQUIRED)
    private String password;

    @Schema(description = "验证码", example = "1234")
    private String captcha;

    @Schema(description = "自动登录", example = "true")
    private Boolean autoLogin;
}

File: harnax-admin/src/main/kotlin/com/agnetix/harnax/admin/dto/LoginRequest.kt
package com.agnetix.harnax.admin.dto

import io.swagger.v3.oas.annotations.media.Schema

/**
 * Login request DTO
 */
@Schema(description = "Login request object")
data class LoginRequest(
    @Schema(description = "Username", example = "admin", requiredMode = Schema.RequiredMode.REQUIRED)
    val username: String = "",

    @Schema(description = "Password", example = "admin123", requiredMode = Schema.RequiredMode.REQUIRED)
    val password: String? = null,

    @Schema(description = "Captcha", example = "1234")
    val captcha: String? = null,

    @Schema(description = "Captcha key, returned when getting captcha", example = "uuid-xxx-xxx")
    val captchaKey: String? = null,

    @Schema(description = "Auto login", example = "true")
    val autoLogin: Boolean? = null,
)
